public/components/GeoPlot.vue

```diff
@@ -976,11 +1178,11 @@ export default Vue.extend({
       // clear polygons
       this.overlay.clearQuads();
       // create quads from latlng
-      const normalizedQuads = this.areaToQuads();
+      const quads = this.currentState.quads();
       // get bounds of quad set
-      const mapBounds = this.getBounds(normalizedQuads);
+      const mapBounds = this.getBounds(quads);
       // add the batched quads to a single layer on the overlay
-      this.overlay.addQuad(this.polygonLayerId, this.areaToQuads());
+      this.overlay.addQuad(this.polygonLayerId, quads);
       // fit map to the quad set
       this.map.fitToBounds(mapBounds);
```

Comment:
Why do you use `map.fitToBounds()` on `onNewData()` but not on `updateMap()`?

Reply:
onNewData gets triggered when new data is being passed in such as highlighting. The map fits the new data to screen to all is visible. updateMap is used when states change from cluster to tile and the map needs to update the renderer. I will add comments for the functions.